### ISSUE TITLE
CAdvisor Instant: "Exited (255)" Status on Amazon Linux AMI

### ISSUE THREAD

**Issue by bmouthrob** [google/cadvisor#1591]
I have looked through the previous issues and it seems that this was said to be fixed, but I'm definitely seeing the issue still!! Docker logs output looks like this:

```
I0208 22:08:06.528378       1 storagedriver.go:50] Caching stats in memory for 2m0s
I0208 22:08:06.528672       1 manager.go:140] cAdvisor running in container: "/docker/aeca51a748adbe6c12dc08550ed15024194544b4235fcb66144bee3dbac08334"
W0208 22:08:06.542050       1 manager.go:148] unable to connect to Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp [::1]:15441: getsockopt: connection refused
I0208 22:08:06.566575       1 fs.go:116] Filesystem partitions: map[/dev/mapper/docker-202:1-395786-98d5d321dc375464f3fb4cdc61dfc09e97c813ce39fe7bde32c1cdef15a7f7b7:{mountpoint:/ major:253 minor:13 fsType:xfs blockSize:0} /dev/xvda1:{mountpoint:/var/lib/docker/devicemapper major:202 minor:1 fsType:ext4 blockSize:0}]
I0208 22:08:06.570021       1 info.go:47] Couldn't collect info from any of the files in "/etc/machine-id,/var/lib/dbus/machine-id"
I0208 22:08:06.570107       1 manager.go:195] Machine: {NumCores:2 CpuFrequency:3000000 MemoryCapacity:16037875712 MachineID: SystemUUID:EC2FE94F-E16E-29BC-DC54-897C42293EDC BootID:e22f450d-be25-4472-a042-b290c9e76394 Filesystems:[{Device:/dev/mapper/docker-202:1-395786-98d5d321dc375464f3fb4cdc61dfc09e97c813ce39fe7bde32c1cdef15a7f7b7 Capacity:10725883904 Type:vfs Inodes:10484736 HasInodes:true} {Device:/dev/xvda1 Capacity:8318783488 Type:vfs Inodes:524288 HasInodes:true}] DiskMap:map[253:0:{Name:dm-0 Major:253 Minor:0 Size:107374182400 Scheduler:none} 253:3:{Name:dm-3 Major:253 Minor:3 Size:10737418240 Scheduler:none} 253:7:{Name:dm-7 Major:253 Minor:7 Size:10737418240 Scheduler:none} 253:8:{Name:dm-8 Major:253 Minor:8 Size:10737418240 Scheduler:none} 253:9:{Name:dm-9 Major:253 Minor:9 Size:10737418240 Scheduler:none} 202:0:{Name:xvda Major:202 Minor:0 Size:8589934592 Scheduler:noop} 253:10:{Name:dm-10 Major:253 Minor:10 Size:10737418240 Scheduler:none} 253:6:{Name:dm-6 Major:253 Minor:6 Size:10737418240 Scheduler:none} 253:12:{Name:dm-12 Major:253 Minor:12 Size:10737418240 Scheduler:none} 253:4:{Name:dm-4 Major:253 Minor:4 Size:10737418240 Scheduler:none} 253:1:{Name:dm-1 Major:253 Minor:1 Size:10737418240 Scheduler:none} 253:11:{Name:dm-11 Major:253 Minor:11 Size:10737418240 Scheduler:none} 253:13:{Name:dm-13 Major:253 Minor:13 Size:10737418240 Scheduler:none} 253:2:{Name:dm-2 Major:253 Minor:2 Size:10737418240 Scheduler:none} 253:5:{Name:dm-5 Major:253 Minor:5 Size:10737418240 Scheduler:none}] NetworkDevices:[{Name:eth0 MacAddress:12:18:27:50:94:b8 Speed:0 Mtu:9001}] Topology:[{Id:0 Memory:16037875712 Cores:[{Id:0 Threads:[0 1] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:47185920 Type:Unified Level:3}]}] CloudProvider:AWS InstanceType:r4.large InstanceID:i-0176c03a9ed8c4cb3}
I0208 22:08:06.570741       1 manager.go:201] Version: {KernelVersion:4.4.41-36.55.amzn1.x86_64 ContainerOsVersion:Alpine Linux v3.4 DockerVersion:1.12.6 CadvisorVersion:v0.24.1 CadvisorRevision:ae6934c}
E0208 22:08:06.580310       1 factory.go:291] devicemapper filesystem stats will not be reported: unable to find thin_ls binary
I0208 22:08:06.580328       1 factory.go:295] Registering Docker factory
W0208 22:08:06.580342       1 manager.go:244] Registration of the rkt container factory failed: unable to communicate with Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp [::1]:15441: getsockopt: connection refused
I0208 22:08:06.580349       1 factory.go:54] Registering systemd factory
I0208 22:08:06.580781       1 factory.go:86] Registering Raw factory
I0208 22:08:06.581134       1 manager.go:1082] Started watching for new ooms in manager
W0208 22:08:06.581430       1 manager.go:272] Could not configure a source for OOM detection, disabling OOM events: unable to find any kernel log file available from our set: [/var/log/kern.log /var/log/messages /var/log/syslog]
I0208 22:08:06.581769       1 manager.go:285] Starting recovery of all containers
I0208 22:08:06.581837       1 manager.go:290] Recovery completed
F0208 22:08:06.581865       1 cadvisor.go:151] Failed to start container manager: inotify_add_watch /var/lib/docker/devicemapper/mnt/98d5d321dc375464f3fb4cdc61dfc09e97c813ce39fe7bde32c1cdef15a7f7b7/rootfs/sys/fs/cgroup/cpu: no such file or directory
```

It really doesn't look like a happy bunny. Any ideas? Thanks.

**Comment by timstclair** [comment]
Duplicate of https://github.com/google/cadvisor/issues/1401

**Comment by bmouthrob** [comment]
Are you sure? I also tried the same with cadvisor-canary:latest image and got the same outcome. Should canary not have picked up this fix yet?

**Comment by timstclair** [comment]
The suggested fix is in the version you're running (v24.1), but based on your log, I don't think it worked :disappointed: 

The symptoms look identical, so I suggest continuing the discussion on the linked issue.

**Comment by bmouthrob** [comment]
I see! Thanks for your time.
